android/libraries/rib-base/src/main/java/com/badoo/ribs/core/routing/backstack/BackStackManager.kt

```diff
@@ -66,7 +64,7 @@ internal class BackStackManager<C : Parcelable>(
     ) : Bootstrapper<Action<C>> {
         override fun invoke(): Observable<Action<C>> = when {
             state.backStack.isEmpty() -> just(Execute(NewRoot(initialConfiguration)))
-            else -> just(ActivateLastEntry())
+            else -> empty()
```

Comment:
Is it restoring state correctly?
How the latest entry is preserved?

Reply:
Check `Router.onAttachView` that calls to `BackStackRibConnector.attachToView()` that essentially does the same as this removed call. Difference is, this removed one was only called after restoring state, while the other is every time view is restored (since view destroyed cleans up last routing action). As soon as the more generic one was implemented, this one became a leftover that doubled the functionality on state restoration and causing the bug.